src/data/ACTIONS/type.ts

```diff
@@ -9,6 +9,34 @@ export type ActionCombo =
 	)
 	& {end?: true}
 
+// Currently whether it's rear or flank doesn't matter.
+export enum BonusModifier {
+	COMBO,
+	POSITIONAL,
+}
+
+// DRG lance mastery is a special case hidden field that
+// augments the potency of the 5th hit.
+export enum PotencySpecialCase {
+	DRG_LANCE_MASTERY,
+}
+
+// When calculating the bonusPercent that the game uses to display
+// combo and positional success, BonusModifiers are the modifiers that
+// will increase the percent. Statuses from jobs like RPR and DRG
+// modify the base.
+export type BaseModifier = PotencySpecialCase | keyof StatusRoot
+
+// Potency is modeled this way because any single potency number
+// can have a combination of states that apply to it, see all
+// of the commented out PotencyModifiers. An empty modifier
+// list means it's the base potency.
+export interface Potency {
+	value: number,
+	baseModifiers?: BaseModifier[]
+	bonusModifiers: BonusModifier[]
+}
```

Comment:
Prefer to not double up usage of a field name and class name like this. Maybe `PotencyConfig` or something instead. That said, I'm not sold on this model yet. A lot of the states overlap so defining discrete states seems weird even if more intuitive than having to calculate the specific modifiers. I'd prob rename `value` here to `potency` and avoid duplicating the existing key name. I'll note that the actual data fields allow any additional stuff so potency itself isn't strictly defined.

Reply:
I don't know what you mean by Strictly Defined. I don't _love_ value and I think potency is better but I don't think "config" makes sense as a name for this object. Potency is close to the right name but a more verbose attempt would be ConditionalPotency, PotencyCondition, or ModifiedPotency (even though it's possibly unmodified). Those are just off the top of my head to be more descriptive, but I think they're worse overall.

Even though I think the `value` field should ideally be called potency, the new Action field `potencies` has a list of `Potency` objects with this naming scheme, which works well.